OpenRA.Mods.Common/AudioLoaders/VocLoader.cs

```diff
@@ -95,11 +96,23 @@ public VocFormat(Stream stream)
 		{
 			this.stream = stream;
 
-			CheckVocHeader();
-			Preload();
+			CheckVocHeader(stream);
+			int sampleRate;
+			Preload(stream, out blocks, out totalSamples, out sampleRate);
+			SampleRate = sampleRate;
+			Rewind();
+		}
+
+		VocFormat(VocFormat cloneFrom)
+		{
+			SampleRate = cloneFrom.SampleRate;
+			stream = SegmentStream.CreateWithoutOwningStream(cloneFrom.stream, 0, (int)cloneFrom.stream.Length);
+			blocks = cloneFrom.blocks;
+			totalSamples = cloneFrom.totalSamples;
+			Rewind();
 		}
```

Comment:
`VocFormat` already implemented the hard part of providing a stream, all we really need to do is to be able to clone it so we can hand out as many as needed on demand.

The cloning is all done here, the main bit is that we create a clone of the source stream, so we don't mess with the original.

The rest of the changes in here are just designed to make it more straightforward to see the cloning is correct. For example, but marking some of the fields as `readonly` - these represent the fixed data that doesn't change. The three remaining non-readonly fields represent the stream position, and are all reset in the `Rewind` method.